### PR TITLE
[lua/cn] synchronized with EN version.

### PR DESCRIPTION
**Pull request by qinhanlei**
1. fib() fixed.
2. add 'local' in `LoudDog:new`.
3. more explain of self-reference.
4. enable Markdown syntax highlight of section `4. Modules.`
5. completed translation of the end.

### CODE REVIEW

File: zh-cn/lua-cn.html.markdown [hunk 1]
```diff
@@ -411,4 +415,9 @@ lua-users.org上的[Lua简明参考](http://lua-users.org/files/wiki_insecure/us
 * <a href="http://lua-users.org/wiki/IoLibraryTutorial">io library</a>
 * <a href="http://lua-users.org/wiki/OsLibraryTutorial">os library</a>
 
+顺便说一下，整个文件是可运行的Lua; 
+保存为 learn-cn.lua 用命令 `lua learn.lua` 启动吧！
+
+本文首次撰写于 [tylerneylon.com](http://tylerneylon.com) 同时也有 <a href="https://gist.github.com/tylerneylon/5853042">github gist</a> 版.
```
Comment (geoffliu) [inline zh-cn/lua-cn.html.markdown]: 链接为什么不用markdown语法？

**Comment by geoffliu** [comment]
多谢多谢。一点小问题，帮忙改一下？:)

**Comment by qinhanlei** [comment]
OK, I'll send a new one.

**Comment by qinhanlei** [comment]
貌似不用关闭这里开辟 [新的PR](https://github.com/adambard/learnxinyminutes-docs/pull/2113)，这里直接就会出现修订的信息... 
囧，PR 的玩法还不熟。

**Pull request closed without merging**